addons/spreadsheet_account/models/account.py

```diff
@@ -58,16 +58,17 @@ def _build_spreadsheet_formula_domain(self, formula_params):
             ("date", ">=", start),
             ("date", "<=", end),
         ]
+        # It is more optimized to (like) search for code directly in account.account than in account_move_line
         code_domain = expression.OR(
             [
-                expression.AND([
-                    [("account_id.code", "=like", f"{code}%")],
-                    expression.OR([balance_domain, pnl_domain]),
-                ])
-                for code in codes
+                ("code", "=like", f"{code}%"),
             ]
+            for code in codes
         )
-        domain = expression.AND([code_domain, [("company_id", "=", company_id)]])
+        full_codes = self.env["account.account"].search(code_domain).mapped("code")
+        code_domain = [("account_id.code", "in", full_codes)]
```

Comment:
actually, we could even go further and check the account ids, right ?
```suggestion
        account_ids = self.env["account.account"].search(code_domain).ids
        code_domain = [("account_id", "in", accounts_ids)]
```

Reply:
Definitely !